client/resteasy-client/src/main/java/com/cerner/beadledom/client/resteasy/ExponentialBackoffHttpRequestRetryHandler.java

```diff
@@ -0,0 +1,45 @@
+package com.cerner.beadledom.client.resteasy;
+
+import java.io.IOException;
+import java.util.Random;
+import org.apache.http.impl.client.StandardHttpRequestRetryHandler;
+import org.apache.http.protocol.HttpContext;
+
+/**
+ * A {@link StandardHttpRequestRetryHandler} that adds an exponential back-off to retries.
+ *
+ * @author John Leacox
+ * @since 3.5
+ */
+class ExponentialBackoffHttpRequestRetryHandler extends StandardHttpRequestRetryHandler {
+  private static final int INITIAL_DELAY = 100;
+  private static final int BACKOFF_FACTOR = 2;
+  private static final Random random = new Random();
+
+  ExponentialBackoffHttpRequestRetryHandler(int retryCount, boolean requestSentRetryEnabled) {
+    super(retryCount, requestSentRetryEnabled);
+  }
+
+  @Override
+  public boolean retryRequest(IOException exception, int executionCount, HttpContext context) {
+    // Adding the delay here is not ideal, but there doesn't appear to be a better place in Apache
+    // httpclient similar to ServiceUnavailableRetryStrategy#getRetryInterval.
+    try {
+      Thread.sleep(calculateRetryDelay(executionCount));
```

Comment:
Don't you want to call `super.retryRequest(exception, executionCount, context);` before sleeping, and only sleep if the request should be retried?

Reply:
Yes, good point. https://github.com/cerner/beadledom/pull/173/commits/e421982ad45b6c0f606cbd8b4f76bbc8c14d5827#diff-2966b7f567e51e0636255711fa0596e7R25